Mathlib/Algebra/Algebra/Defs.lean

```diff
@@ -311,6 +324,47 @@ section
 
 end
 
+section compHom
+variable {R : Type u} {S : Type v} (A : Type w)
+    [CommSemiring R] [Semiring A] [Algebra R A] [CommSemiring S]
+    (f : S →+* R)
+
+/--
+Compose an `Algebra` with a `RingHom`, with action `f s • m`.
+
+This is the algebra version of `Module.compHom`.
+-/
+abbrev compHom : Algebra S A where
```

Comment:
Can you please motivate why you want this definition? (Maybe you already did that, and I missed it.)

Since you need all those `letI`s below, I think it will be somewhat hard to use in practice. If there is a good use case, it should probably be explained in the documentation.

Reply:
I think "we should have this because we already have `Module.compHom`" is sufficient to motivate this definition, though elsewhere we don't bother with any lemmas about it.